signingscript/src/signingscript/sign.py

```diff
@@ -1523,11 +1510,36 @@ def _can_notarize(filename, supported_extensions):
     return extension in supported_extensions
 
 
+async def _notarize_single(path, creds_path):
+    """Notarizes a single app/pkg retrying if necessary"""
+    ATTEMPTS = 5
+    # Notarize
+    submission_id = await retry_async(
+        func=rcodesign_notarize,
+        args=(path, creds_path),
+        attempts=ATTEMPTS,
+        retry_exceptions=RCodesignError,
+    )
+    # Wait for notary to be done
+    await retry_async(
+        func=rcodesign_notary_wait,
+        args=(submission_id, creds_path),
+        attempts=ATTEMPTS,
+        retry_exceptions=RCodesignError,
+    )
+    # Staple
+    await retry_async(
+        func=rcodesign_staple,
+        args=[path],
+        attempts=ATTEMPTS,
+        retry_exceptions=RCodesignError,
+    )
```

Comment:
Are you sure we need to call all of these? The `rcodesign` docs seems to suggest that your first function should be waiting and stapling all by itself. It says: "Or to wait and automatically staple the file if notarization was successful:"
```
rcodesign notary-submit \
  --api-key-path ~/.appstoreconnect/key.json \
  --staple \
  path/to/file/to/notarize
```

It seems like the first call being retried ought to do everything we need (at the cost of resubmitting entirely on _any_ failure).

I think a better way to approach this may be:
1) Submit without `--staple`, so that `notarize` returns right away. This should be retried because the submission may fail
2) Run `notary_wait` with retries (obviously the polling may fail for various reasons).
3) Run `staple`. There shouldn't be any need to retry here.

What do you think? Am I overlooking something?

Reply:
`staple` does reach out to the Apple servers to collect the ticket, therefore I see the benefit of having a retry. But I can change it if you still think it's a good idea.

Reply:
The way I changed it "wait and staple" is optional. In this case, it just returns the submission id, then we have the wait, and then stapling.
```
# function:
async def rcodesign_notarize(app_path, creds_path, staple=False):
.......
# usage:
    submission_id = await retry_async(
        func=rcodesign_notarize,
        args=(path, creds_path), # <--- no staple
```
So, the first command only submits and returns the submission id

Reply:
Oh right, sorry. I somehow glossed over the fact that `--staple` was optional, heh. Alright, proceed!